src/Avalonia.Base/Media/TextFormatting/CharacterBufferRange.cs

```diff
@@ -140,19 +140,19 @@ internal CharacterBufferRange(TextRun textRun)
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
 #endif
-                return Span[index];
+                return CharacterBufferReference.CharacterBuffer.Span[CharacterBufferReference.OffsetToFirstChar + index];
             }
         }
 
         /// <summary>
         /// Gets a reference to the character buffer
         /// </summary>
-        public CharacterBufferReference CharacterBufferReference { get; }
+        public readonly CharacterBufferReference CharacterBufferReference;
```

Comment:
I'm pretty sure that JIT optimizes primitive getter usage.

Reply:
It might have effect on inlining other getters, so I'd suggest to just split fields and properties, so other class members would use direct field access.

Reply:
Can we verify that this change has an effect on performance? If so other objects would suffer from the same issue and we need to adjust them.

Reply:
If we just need to add a backing field to fix this I would prefer not having a public field

Reply:
Making fields from properties - improved performance a bit. that's why it's in commit. all changes that did not affect perf was rolled back and do not exists here.
I can't give a strong answer why. If you wish I can rollback this changes. No problem. Other changes have more impact.

Reply:
I don't want to take your and my time investigating this minor size questions. Just did a rollback.